include/SFML/System/Vector2.inl

```diff
@@ -159,3 +159,24 @@ constexpr bool operator !=(const Vector2<T>& left, const Vector2<T>& right)
 {
     return (left.x != right.x) || (left.y != right.y);
 }
+
+////////////////////////////////////////////////////////////
+template <typename T>
+constexpr T operator *(const Vector2<T>& left, const Vector2<T>& right)
+{
+    return (left.x * right.x) + (left.y * right.y);
+}
+
+////////////////////////////////////////////////////////////
+template<typename T>
+constexpr T cross(Vector2<T>& left, Vector2<T>& right)
+{
+    return (left.x*right.y) - (left.y*right.x);
```

Comment:
Add spaces around asterisks. See other SFML .cpp files to get a better idea for formatting conventions and be sure to match those conventions as well as possible.